client/my-sites/stats/stats-date-picker/index.jsx

```diff
@@ -171,8 +173,10 @@ class StatsDatePicker extends Component {
 
 	render() {
 		/* eslint-disable wpcalypso/jsx-classname-namespace*/
-		const { summary, translate, query, showQueryDate, isActivity, isShort } = this.props;
+		const { summary, translate, query, showQueryDate, isActivity, isShort, dateRange, reduxState } =
+			this.props;
 		const isSummarizeQuery = get( query, 'summarize' );
+		const { selectedShortcut } = getShortcuts( reduxState, dateRange, undefined, translate );
```

Comment:
```suggestion
		const { selectedShortcut } = getShortcuts( reduxState, dateRange, undefined, translate, isNewDateFilteringEnabled );
```
We might need to pass `isNewDateFilteringEnabled` from the [ancestor](https://github.com/Automattic/wp-calypso/blob/26dbfeaedd1434a405b695b9980baf4e3e2f795e/client/my-sites/stats/site.jsx#L485-L493) for internal sites check for Today and Yesterday.

<img width="1105" alt="截圖 2024-12-02 下午2 45 29" src="https://github.com/user-attachments/assets/abbf139c-6eeb-4242-8a20-67e249d19acc">

Reply:
Good catch!